File: examples/05-bullet-styles/main.go
// examples/05-bullet-styles/main.go demonstrates all available bullet list styles.
//
// Shows standard bullets, numbered lists, lettered lists, and sub-bullets at
// various indent levels using the SlideContent fluent API.
//
// Run with: go run ./examples/05-bullet-styles/main.go
package main

import (
	"fmt"
	"os"
	"path/filepath"

	log "github.com/djinn-soul/gopptx/pkg/stdlog"

	"github.com/djinn-soul/gopptx/pkg/pptx"
)

const (
	outputDir  = "examples/output"
	outputFile = "05_bullet_styles.pptx"
)

func main() {
	if err := run(); err != nil {
		fmt.Fprintf(os.Stderr, "error: %v\n", err)
		os.Exit(1)
	}
}

func run() error {
	if err := os.MkdirAll(outputDir, 0o750); err != nil {
		return fmt.Errorf("create output directory: %w", err)
	}

	// Slide 1: Standard bullet points
	slideStandard := pptx.NewSlide("Standard Bullet Points").
		AddBullet("First standard bullet item.").
		AddBullet("Second standard bullet item.").
		AddBullet("Third standard bullet item.")

	// Slide 2: Numbered bullet list
	slideNumbered := pptx.NewSlide("Numbered List").
		AddNumbered("First numbered item.").
		AddNumbered("Second numbered item.").
		AddNumbered("Third numbered item.")

	// Slide 3: Lettered bullet list
	slideLettered := pptx.NewSlide("Lettered List").
		AddLettered("Item alpha.").
		AddLettered("Item beta.").
		AddLettered("Item gamma.")

	// Slide 4: Sub-bullets at multiple indent levels
	slideSubBullets := pptx.NewSlide("Sub-Bullet Levels").
		AddBullet("Top-level bullet (level 0).").
		AddSubBullet(1, "Sub-bullet at level 1.").
		AddSubBullet(2, "Sub-bullet at level 2.").
		AddBullet("Back to top-level.").
		AddSubBullet(1, "Another level 1 sub-bullet.")

	// Slide 5: Mixed styles on one slide
	slideMixed := pptx.NewSlide("Mixed Bullet Styles").
		AddBullet("Standard bullet item.").
		AddNumbered("Numbered item follows.").
		AddLettered("Lettered item next.").
		AddSubBullet(1, "Indented sub-point.")

	slides := []pptx.SlideContent{
		slideStandard,
		slideNumbered,
		slideLettered,
		slideSubBullets,
		slideMixed,
	}

	data, err := pptx.CreateWithSlides("Bullet Styles Demo", slides)
	if err != nil {
		return fmt.Errorf("create presentation: %w", err)
	}

	outputPath := filepath.Join(outputDir, outputFile)
	if err = os.WriteFile(outputPath, data, 0o600); err != nil {
		return fmt.Errorf("write file: %w", err)
	}

	log.Printf("Generated %s\n", outputPath)
	return nil
}

File: examples/05-bullet-styles/main.py
"""Demonstrate all available bullet list styles.

This example demonstrates:
- Standard bullet points
- Numbered (ordered) lists
- Lettered lists
- Sub-bullets at multiple indent levels
- Mixed bullet styles on one slide
"""

from __future__ import annotations

from pathlib import Path

from gopptx import Presentation


def _add_standard_bullets(prs: Presentation) -> None:
    """Add a slide with standard bullet points."""
    prs.add_bullet_slide(
        "Standard Bullet Points",
        [
            "First standard bullet item.",
            "Second standard bullet item.",
            "Third standard bullet item.",
        ],
    )


def _add_numbered_list(prs: Presentation) -> None:
    """Add a slide with a numbered list."""
    prs.add_bullet_slide(
        "Numbered List",
        [
            "1. First numbered item.",
            "2. Second numbered item.",
            "3. Third numbered item.",
        ],
    )


def _add_lettered_list(prs: Presentation) -> None:
    """Add a slide with a lettered list."""
    prs.add_bullet_slide(
        "Lettered List",
        [
            "a. Item alpha.",
            "b. Item beta.",
            "c. Item gamma.",
        ],
    )


def _add_sub_bullets(prs: Presentation) -> None:
    """Add a slide with sub-bullets at multiple indent levels."""
    prs.add_bullet_slide(
        "Sub-Bullet Levels",
        [
            "Top-level bullet (level 0).",
            "  Sub-bullet at level 1.",
            "    Sub-bullet at level 2.",
            "Back to top-level.",
            "  Another level 1 sub-bullet.",
        ],
    )


def _add_mixed_styles(prs: Presentation) -> None:
    """Add a slide mixing different bullet styles."""
    prs.add_bullet_slide(
        "Mixed Bullet Styles",
        [
            "Standard bullet item.",
            "1. Numbered item follows.",
            "a. Lettered item next.",
            "  Indented sub-point.",
        ],
    )


def main() -> None:
    """Create a presentation demonstrating all bullet list styles."""
    output_dir = Path("examples/output")
    output_dir.mkdir(exist_ok=True)

    with Presentation.new("Bullet Styles Demo") as prs:
        _add_standard_bullets(prs)
        _add_numbered_list(prs)
        _add_lettered_list(prs)
        _add_sub_bullets(prs)
        _add_mixed_styles(prs)

        output_path = output_dir / "05-bullet-styles.pptx"
        prs.save(str(output_path))
        print(f"Saved: {output_path}")

    print("\n=== SUMMARY ===")
    print("Created 5 slides demonstrating bullet list styles:")
    print("  - Standard bullets")
    print("  - Numbered list")
    print("  - Lettered list")
    print("  - Sub-bullets at multiple levels")
    print("  - Mixed styles")


if __name__ == "__main__":
    main()
